index.js

```diff
@@ -232,13 +126,17 @@ Protocol.prototype._emitData = function _emitData(chunk){
 };
 
 Protocol.prototype.listenPort = function listenPort(cb){
-  this._transport.on('data', this._emitData.bind(this));
+  this.on('data', this._emitData.bind(this));
 
   return nodefn.bindCallback(when.resolve(), cb);
 };
 
+Protocol.prototype._write = function write(data){
```

Comment:
why expose a `write` method, you can just call `transport.send` where it is needed.  Less API surface